nipype/interfaces/mrtrix3/utils.py

```diff
@@ -1186,3 +1186,203 @@ def _list_outputs(self):
         outputs = self.output_spec().get()
         outputs["out_file"] = op.abspath(self.inputs.out_file)
         return outputs
+
+class MaskFilterInputSpec(CommandLineInputSpec):
+    in_file = File(
+        exists=True,
+        mandatory=True,
+        argstr="%s",
+        position=-3,
+        desc="Input mask",
+    )
+    filter = traits.Str(
+        mandatory=True, 
+        argstr="%s",
+        position=-2,
+        desc="Filter to perform (e.g. dilate, erode)"
+    )
+    out_file = File(
+        name_source=["input_image"],
+        mandatory=True,
+        argstr="%s",
+        position=-1,
+        desc="Output mask"
+    )
+    npass = traits.Int(
+        argstr="-npass %d",
+        position=1,
+        desc="Number of passes"
+    )
+
+class MaskFilterOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc="the filtered output mask")
+
+class MaskFilter(CommandLine):
+    """
+    Perform filtering operations on 3D / 4D mask images. 
+    Only supports dilate / erode filters at the moment.
+    For more information see: https://mrtrix.readthedocs.io/en/latest/reference/commands/maskfilter.html
+    
+
+    Example
+    -------
+
+    >>> import nipype.interfaces.mrtrix3 as mrt
+    >>> mf = mrt.MaskFilter()
+    >>> mf.inputs.in_file = 'mask.mif'
+    >>> mf.inputs.filter = 'dilate'
+    >>> mf.inputs.npass = 2
+    >>> mf.out_file = 'mask_filtered.mif'
+    >>> mf.cmdline
+    'maskfilter -npass 2 mask.mif dilate mask_filtered.mif'
+    >>> mf.run()
+    """
+
+    _cmd = "maskfilter"
+    input_spec = MaskFilterInputSpec
+    output_spec = MaskFilterOutputSpec
+        
+    def _list_outputs(self):
+        outputs = self.output_spec().get()
+        outputs["out_file"] = op.abspath(self.inputs.out_file)
+        return outputs
+    
+class MTNormaliseInputSpec(MRTrix3BaseInputSpec):
+    wm_fod = File(
+        argstr="%s",
+        exists=True,
+        mandatory=False,
+        position=1,
+        desc="input fod of white matter tissue compartment"
+    )
+    out_file_wm = File(
+        argstr="%s",
+        mandatory=False,
+        position=2,
+        desc="output file of white matter tissue compartment"
+    )
+    gm_fod = File(
+        argstr="%s",
+        exists=True,
+        mandatory=False,
+        position=3,
+        desc="input fod of grey matter tissue compartment"
+    )
+    out_file_gm = File(
+        argstr="%s",
+        mandatory=False,
+        position=4,
+        desc="output file of grey matter tissue compartment"
+    )
+    csf_fod = File(
+        argstr="%s",
+        exists=True,
+        mandatory=False,
+        position=5,
+        desc="input fod of CSF tissue compartment"
+    )
+    out_file_csf = File(
+        argstr="%s",
+        mandatory=False,
```

Comment:
`mandatory=False` is default. If something is mandatory, add `mandatory=True`, otherwise omit.

```suggestion
        position=1,
        desc="input fod of white matter tissue compartment"
    )
    out_file_wm = File(
        argstr="%s",
        position=2,
        desc="output file of white matter tissue compartment"
    )
    gm_fod = File(
        argstr="%s",
        exists=True,
        position=3,
        desc="input fod of grey matter tissue compartment"
    )
    out_file_gm = File(
        argstr="%s",
        position=4,
        desc="output file of grey matter tissue compartment"
    )
    csf_fod = File(
        argstr="%s",
        exists=True,
        position=5,
        desc="input fod of CSF tissue compartment"
    )
    out_file_csf = File(
        argstr="%s",
```